src/test/ui/consts/trait_specialization.rs

```diff
@@ -0,0 +1,64 @@
+// ignore-wasm32-bare which doesn't support `std::process:exit()`
+// run-pass
+
+// Tests that specialization does not cause optimizations running on polymorphic MIR to resolve
```

Comment:
Don't you also need `// compile-flags: -Zmir-opt-level=3`?

Reply:
To trigger the inliner? No, it runs on `mir-opt-level=2`

Reply:
Ok, then `=2` :D, since ui tests don't run optimizations afaik

Reply:
Oh duh. I wrote this originally as a `mir-opt` test but it was going to be like 800 lines of output and I didn't think it was worth it.